#373 Avatar component

sgraczyk opened the pull request:
Resolves: #373 

## Description
Implements the `Avatar` component from scratch. During the development phase I made some decisions to be confirmed by the rest of the team:
* I decided to make the API more generic. For example instead of `kind: 'person' | 'group'` I've used `shape: 'circle' : 'rounded-square'`. A similar change was for the status, as not all products are chat-focused, so I avoided the `accepts chats` name.
* Figma project does not specify the behavior of rim in case of `rounded-square`, nevertheless, I implemented it.
* I'm a bit confused on how to address the position of the status icon as the design seems to do it differently for each size and square. I tried to follow that assumption with some slight modifications.

## Storybook
- [Avatar](https://feature-373--613a8e945a5665003a05113b.chromatic.com/?path=/story/components-avatar--default)

## Checklist

**Obligatory:**

- [x] Self-review
- [ ] Unit & integration tests
- [x] Storybook cases
- [x] Design review by @kstvsko 
- [x] Functional (QA) review by @TomaszRatajczyk 

**Optional:**

- [ ] Accessibility cases (keyboard control, correct HTML markup, etc.)

sgraczyk commented on packages/react-components/src/components/Avatar/Avatar.helpers.ts:
```diff
@@ -0,0 +1,16 @@
+import { getContrast } from 'polished';
+
+export function getInitials(name = '', count = 2): string {
```
At this moment it does not support diacritics or emojis.

kstvsko commented:
@sgraczyk 

I've added updated avatars, also with status indicators, that are now scaled and positioned at the same place at every size. I've added also the scale for the avatars with rims. :) 

Rims: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=8167%3A31183
Status Indicators: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=736%3A8527

MichalPaszowski commented:
Something not pykło here https://monosnap.com/file/1aOi1ZZ2lR00rwnWGF2JmDl9Ks13r3

sgraczyk commented:
> Something not pykło here https://monosnap.com/file/1aOi1ZZ2lR00rwnWGF2JmDl9Ks13r3

Should be fixed within https://github.com/livechat/design-system/pull/375/commits/99b29a141f0edf68f0946b178b36731967670c83

sgraczyk commented:
> @sgraczyk
> 
> I've added updated avatars, also with status indicators, that are now scaled and positioned at the same place at every size. I've added also the scale for the avatars with rims. :)
> 
> Rims: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=8167%3A31183 Status Indicators: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=736%3A8527

Status indicators are implemented as proposed, however, I'd like to wait before introducing changes designed for rims. It looks that now the size of the image is adjusted so the rim fits into the original size of the avatar. It'll probably introduce a flickering effect, especially that the rim is a UI element that may appear and disappear when a specific condition changes. Can we move the work around rims to another issue, and accept the current state for now?

MichalPaszowski commented on packages/react-components/src/components/Avatar/Avatar.spec.tsx:
```diff
@@ -0,0 +1,139 @@
+    'xxxsmall',
+    'xxsmall',
+    'xsmall',
+  ];
```
missing space - perhaps it should be put up top of the file?

MichalPaszowski commented on packages/react-components/src/components/Avatar/Avatar.spec.tsx:
```diff
@@ -0,0 +1,139 @@
+    'xlarge',
+    'xxlarge',
+  ];
+  it.each(smallAndAboveSizes)(
```
same as above
